plugins/woocommerce/includes/admin/class-wc-admin-assets.php

```diff
@@ -588,11 +593,20 @@ public function enqueue_block_editor_assets() {
 				$analytics_reports = Analytics::get_report_pages();
 				if ( is_array( $analytics_reports ) && count( $analytics_reports ) > 0 ) {
 					$formatted_analytics_reports = array_map( function( $report ) {
-						return array(
-							'title' => is_string( $report['title'] ) ? $report['title']: '' ,
-							'path' => is_string( $report['path'] ) ? $report['path']: '',
-						);
+						$title = array_key_exists( 'title', $report ) ? $report['title'] : '';
+						$path = array_key_exists( 'path', $report ) ? $report['path'] : '';
+						if (
+							is_string( $title ) && $title !== "" &&
+							is_string( $path ) && $path !== ""
+						) {
+							return array(
+								'title' => $title,
+								'path' => $path,
+							);
+						}
+						return null;
 					}, $analytics_reports );
+					$formatted_analytics_reports = array_filter( $formatted_analytics_reports, 'is_array' );
```

Comment:
So `array_filter` should be on line 608 `array_filter( $analytics_reports );` before you use it and line 609 is not needed. Otherwise you'll get:
```
Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /Users/royho/projects/woocommerce-monorepo/plugins/woocommerce/includes/admin/class-wc-admin-assets.php:596 Stack trace: #0 [internal function]: WC_Admin_Assets->{closure}(NULL) #1 /Users/royho/projects/woocommerce-monorepo/plugins/woocommerce/includes/admin/class-wc-admin-assets.php(608): array_map(Object(Closure), Array) #2 /Users/royho/Projects/woo-blocks-dev/app/public/wp-includes/class-wp-hook.php(324): WC_Admin_Assets->enqueue_block_editor_assets('') #3 /Users/royho/Projects/woo-blocks-dev/app/public/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array) #4 /Users/royho/Projects/woo-blocks-dev/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #5 /Users/royho/Projects/woo-blocks-dev/app/public/wp-admin/edit-form-blocks.php(272): do_action('enqueue_block_e...') #6 /Users/royho/Projects/woo-blocks-dev/app/public/wp-admin/post-new.php(72): require('/Users/royho/Pr...') #7 {main} thrown in /Users/royho/projects/woocommerce-monorepo/plugins/woocommerce/includes/admin/class-wc-admin-assets.php on line 596
```
On page/post when adding new.

Reply:
Oh, good catch! I see what you mean. I updated the testing steps in the PR and pushed another commit to this PR. I ended up adding the check inside the `array_map()`:

```
if ( ! is_array( $report ) ) {
	return null;
}
```
The rationale is that given that we need to iterate through the array in `array_map()`, I think it's more optimal doing the check there than using `array_filter()` before. But I'm happy to change it if it doesn't make sense.

Related to line 609, I wasn't able to remove it because if the `$report` is an array but has wrong data (missing `title` or `path`) its value will be `null`, so we need to filter it out.